### PR TITLE
right prompt implementation

### PR DESCRIPTION
**Pull request by scopatz**
As requested by @arkhan, here is the right prompt implementation.

### CODE REVIEW

File: xonsh/ptk/shortcuts.py [hunk 1]
```diff
@@ -61,6 +64,8 @@ def prompt(self, message='', **kwargs):
 
         # Create CommandLineInterface.
         if self.cli is None:
+            if self.major_minor <= (0, 57):
```
Comment (gforsyth) [inline xonsh/ptk/shortcuts.py]: Hey @scopatz -- `prompt_toolkit` master still reports itself as version 0.57 so the `get_rprompt_tokens` kwarg never makes it past this conditional.  
I removed it and then `$RIGHT_PROMPT` works as expected.

**Comment by gforsyth** [comment]
As per my comment above, maybe the easiest thing is just to hold this until the next release of `prompt_toolkit`?

**Comment by scopatz** [comment]
> As per my comment above, maybe the easiest thing is just to hold this until the next release of prompt_toolkit?

Sounds good to me.  I think that a release is coming relatively soon.

**Comment by scopatz** [comment]
Note that I have now made this dependent on the `c` branch, due to some conflicts with fixes there.

**Comment by scopatz** [comment]
Hey @gforsyth - ptk v0.58 has been released and the CI failure is from the broken binstar stuff.  Maybe we should just disable this :( @melund, thoughts?

**Comment by gforsyth** [comment]
It would be great if the binstar CI worked, but I don't think we should let it block anything.  Merging this in now.

**Comment by scopatz** [comment]
Thanks @gforsyth! I agree about binstar.  I think I have tracked down the problem to the git version numbering thing.  We should probably just use the release script to fill in version numbers in `meta.yaml`.  It seems more reliable than conda build right now.

**Comment by melund** [comment]
@scopatz . I also think that binstar/anaconda.org is too unreliable. Can you remove it from the CI tests?

I will see what I can do with the git version number stuff. I have also seen some weird behaviour with this. But only on windows. So I thought it was transient bug in conda build on windows. 

I don't want to mix the release script into this. It would just make it difficult to build 'cutting edge'  conda packages from master.

**Comment by scopatz** [comment]
Ok @melund, binstar has been removed.  Let me know if you want me to rollback the change to the recipe that is currently on the `c` branch.  The git version shouldn't get in the way of anything now that the CI is gone.

**Comment by melund** [comment]
Yes. Do that. I suggest you add the following fix on the version line instead. That fixed some git version problem for me: 

```
  version: {{ environ.get('GIT_DESCRIBE_TAG', '0.0.0') }}
```

**Comment by scopatz** [comment]
done!  Nice fix too.

**Pull request merged**